commands/inspect_image.go

```diff
@@ -0,0 +1,147 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpack/pack"
+	"github.com/buildpack/pack/config"
+	"github.com/buildpack/pack/style"
+
+	"github.com/buildpack/pack/logging"
+)
+
+type InspectImageFlags struct {
+	BOM bool
+}
+
+func InspectImage(logger logging.Logger, cfg *config.Config, client PackClient) *cobra.Command {
+	var flags InspectImageFlags
+	cmd := &cobra.Command{
+		Use:   "inspect-image <image-name>",
+		Short: "Show information about a built image",
+		Args:  cobra.ExactArgs(1),
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			img := args[0]
+			remote, err := client.InspectImage(img, false)
+			if err != nil {
+				logger.Errorf("inspecting remote image '%s': %s", img, err)
+			}
+			local, err := client.InspectImage(img, true)
+			if err != nil {
+				logger.Errorf("inspecting local image '%s': %s", img, err)
+			}
+			if flags.BOM {
+				return logBOM(remote, local, logger)
+			}
+			logger.Infof("Inspecting image: %s\n", style.Symbol(img))
+			logDetails(remote, local, *cfg, logger)
+			return nil
+		}),
+	}
+	AddHelpFlag(cmd, "inspect-image")
+	cmd.Flags().BoolVar(&flags.BOM, "bom", false, "print bill of materials")
+	return cmd
+}
+
+type bom struct {
+	Remote interface{} `json:"remote"`
+	Local  interface{} `json:"local"`
+}
+
+func logBOM(remote *pack.ImageInfo, local *pack.ImageInfo, logger logging.Logger) error {
+	rawBOM, err := json.Marshal(bom{
+		Remote: remote.BOM,
+		Local:  local.BOM,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "writing bill of materials")
+	}
+	logger.Infof(string(rawBOM))
+	return nil
+}
+
+func logDetails(remote *pack.ImageInfo, local *pack.ImageInfo, cfg config.Config, logger logging.Logger) {
+	imgTpl := template.Must(template.New("runImages").Parse(runImagesTemplate))
+	imgTpl = template.Must(imgTpl.New("buildpacks").Parse(buildpacksTemplate))
```

Comment:
Does this sort of thing append to the template? Seems strange, but if it works, it works.